### PR TITLE
Show a deposit schedule notice on the deposits list page

### LINKED ISSUE #7877: Deposits listing page no longer shows next deposit date.
### Describe the bug

With the removal of estimated deposits, merchants do not have a clear understanding of when their next deposit is due from the deposit listing page.

To help resolve this, we should add the deposit schedule to the page as either a notice or as a row in the table. This can be a duplication of the information on the payments overview screen.

**Sample Mockup**

image

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #7877

Based on PR from #7875 since it relies on changes to the `DepositSchedule` component.

#### Changes proposed in this Pull Request

Adds a notice to the Deposits List screen to communicate the deposit schedule for the account. The "next deposit" part of the message is **not** rendered here since this screen can include multiple currencies with differing next deposit dates.

This notice will only be rendered if deposits are unrestricted (not blocked or on a restricted schedule).

This notice is dismissable by the user. When this occurs, the dismissal state will persist until the user changes their deposit schedule settings.

https://github.com/Automattic/woocommerce-payments/assets/3147296/4d8e62ef-724e-41b8-b33b-f3b020594be5

#### Testing instructions

* Check that the notice is rendered on the Deposits List screen for accounts without deposit restrictions (use MC tools to add/remove restrictions to account)
* Check that the message is not rendered if
	* Deposits blocked
	* Deposits restricted
* Dismiss the notice and refresh the page. Ensure the notice remains hidden.
* Change your deposit schedule via Payments → Settings.
* Ensure the notice is now visible on the Deposits List screen and shows the correct schedule.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7883` or branch name `add/7877-deposit-list-next-deposit-notice` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/7877-deposit-list-next-deposit-notice) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 5d9efb678fbe17ae0b6639f9ec2f6a11fd246559
- Build time: 2023-12-13 05:52:58 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/globals.d.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * Internal dependencies
  */
-import {
+import type {
```
Comment (shendy-a8c) [inline client/globals.d.ts:4]: I think I asked you this but forgot the reason.

Googled and found https://medium.com/@quizzesforyou/import-vs-import-type-in-typescript-8e5177b62bea

File: client/deposits/index.tsx [hunk 2]
```diff
@@ -10,12 +10,37 @@ import React from 'react';
...
 import DepositsList from './list';
 
+const NextDepositNotice: React.FC = () => {
+	const {
+		overviews: { account },
+	} = useAllDepositsOverviews();
+
+	const depositSettings = wcpaySettings.accountStatus?.deposits;
+	const isDepositsUnrestricted =
+		wcpaySettings.accountStatus.deposits?.restrictions ===
```
Comment (shendy-a8c) [inline client/deposits/index.tsx:49]: Would this work?

```suggestion
		depositSettings?.restrictions ===
```

File: client/deposits/index.tsx [hunk 3]
```diff
@@ -10,12 +10,37 @@ import React from 'react';
...
+		'deposits_unrestricted';
+
+	if ( ! depositSettings || ! isDepositsUnrestricted ) {
```
Comment (shendy-a8c) [inline client/deposits/index.tsx]: When `account` is still being fetched, reference to `account.deposits_schedule` will throw an error.

I got this error on page load.

Screenshot 2023-12-13 at 00 16 57

I think we can fix by returning early here.

```suggestion
	if ( ! depositSettings || ! isDepositsUnrestricted || ! account ) {
```

**Review by shendy-a8c — CHANGES_REQUESTED** [review]

File: client/deposits/index.tsx (hunk 3 above)
Comment (csmcneill) [reply client/deposits/index.tsx]: I'm getting this error when testing on my site, too.

Comment (Jinksi) [reply client/deposits/index.tsx]: Doh, thanks!

File: client/deposits/index.tsx (hunk 2 above)
Comment (Jinksi) [reply client/deposits/index.tsx:49]: Even better, remove `depositSettings`` entirely. I'm not sure why I left that in here.

Resolved in 25de881ab9695ddb149523273cc08a4eee7fa3d8

File: client/globals.d.ts (hunk 1 above)
Comment (Jinksi) [reply client/globals.d.ts:4]: This is my personal preference, I think it is more explicit === improved code readability.

File: client/deposits/index.tsx (hunk 3 above)
Comment (Jinksi) [reply client/deposits/index.tsx]: TS not helping me out here! I'll fix up the return type of `useAllDepositsOverviews()` in a followup PR.

Comment (Jinksi) [reply client/deposits/index.tsx]: Resolved in 8ec00b158c23403c74a7b624ce5793d284ce4d64

**Comment by Jinksi** [comment]
I've now made this notice dismissable by the user in 3ed5c04.

It will no show up again unless the user changes their deposit schedule settings.

See screen recording

https://github.com/Automattic/woocommerce-payments/assets/3147296/4d8e62ef-724e-41b8-b33b-f3b020594be5

**Review by brucealdridge — APPROVED** [review]
Tested and working as expected. LGTM.

**Comment by Jinksi** [comment]
Rebased to `fix/7874-show-deposit-schedule-when-zero-available-balance` and removed all commits carried over from `develop`.

**Comment by brucealdridge** [comment]
@Jinksi it does look like the history is still muddled

**Comment by Jinksi** [comment]
> @Jinksi it does look like the history is still muddled

Yeh it looks like it includes commits from `release/6.9.2` branch and `fix/7874-show-deposit-schedule-when-zero-available-balance`, but I think this is OK since these are pulled into this branch when clicking "Update branch" in GH PR UI.

**Pull request merged**